lazyvim/extras/lang/rust.nix

```diff
@@ -0,0 +1,36 @@
+self:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib.modules) mkIf;
+  inherit (lib.options) mkEnableOption;
+
+  cfg = config.programs.lazyvim;
+in
+{
+  options.programs.lazyvim.extras.lang.rust = {
+    enable = mkEnableOption "the lang.rust extra";
+  };
+
+  config = mkIf cfg.extras.lang.rust.enable {
+    programs.neovim = {
+      extraPackages = [ pkgs.rust-analyzer ];
```

Comment:
Thanks for this PR! I tested it out and would just note that it might be useful to be able to disable installation of rust-analyzer https://github.com/nix-community/nixvim/discussions/1135. Not that it's ideal, but, for example, if someone were using rustup, this could conflict.